src/bech32.c

```diff
@@ -184,10 +202,13 @@ static int segwit_addr_decode(int *witver, uint8_t *witdata, size_t *witdata_len
     uint8_t data[84];
     char hrp_actual[84];
     size_t data_len;
-    if (!bech32_decode(hrp_actual, data, &data_len, addr, 90)) goto fail;
+    bech32_encoding enc = BECH32_ENCODING_NONE;
```

Comment:
declare this with the other vars at the start of the function, and remove the unneeded initialisation.